### DISCUSSION TITLE
Using Timber in a DIVI extension

### DISCUSSION THREAD

**Discussion by robibi78** [timber/timber#2590] [Q&A] [answered]
Hello,

I try to use Timber inside a DIVI extension. Everything is working expect I have to render.

```
 public function render($unprocessed_props, $content, $render_slug) {
    $context = Timber::get_context();
    $context['publications'] = $this->get_hal('FRANCE-GRILLES')->docs;

    return Timber::render( 'page.twig', $context );
  }

```
When I do that, I have one render outside divi extension, and one inside.

Is it possible to have only one render inside the divi extension ? 

Thks

**Comment by gchtr** [comment, marked as answer]
@robibi78 Try using `Timber::compile()` instead of `Timber::render()`. The `Timber::render()` method **echoes the output**. Because you use a return, I assume you don’t want your output to be echoed in your render function, but only returned as a string:

```php
public function render( $unprocessed_props, $content, $render_slug ) {
    $context                 = Timber::get_context();
    $context['publications'] = $this->get_hal( 'FRANCE-GRILLES' )->docs;

    return Timber::compile( 'page.twig', $context );
}
```

**Comment by robibi78** [comment]
It was so easy, thks for the answer :)

**Reply by gchtr** [reply]
You’re welcome :)
